python/pyspark/sql/session.py

```diff
@@ -416,6 +417,50 @@ def _createFromLocal(self, data, schema):
         data = [schema.toInternal(row) for row in data]
         return self._sc.parallelize(data), schema
 
+    def _getNumpyRecordDtypes(self, rec):
```

Comment:
nit: `_getNumpyRecordDtypes` -> `_get_numpy_record_dtypes`.

Reply:
I know it's confusing .. but I usually use thisNamingRule mainly on the purpose of API consistency and otherwise use this_naming_rule. I actually checked and read documentation and other codes few times for clarification for myself .. I believe this_naming_rule is preferred by PEP 8.

But I know that the [doc](https://www.python.org/dev/peps/pep-0008/#function-names) says:

> mixedCase is allowed only in contexts where that's already the prevailing style (e.g. threading.py), to retain backwards compatibility.

but .. I believe we should avoid thisNamingRule if it's in particular for internal use and/or unrelated with compatibility. Up to my knowledge, `threading.py` is the similar case I believe ...

Reply:
yeah, I agree we should be using lowercase with underscores which is more of the convention for python.  I was only using this format to stay consistent with the rest of the file, but I can change it.  Just for the new methods right?